Code_Exercises/helpers.hpp

```diff
@@ -11,16 +11,106 @@
 #pragma once
 
 #include <cstddef>  // for size_t
+#include <type_traits>
+#include <utility>
 
 #ifndef __SYCL_DEVICE_ONLY__
 #include <cstdio>   // fprintf
 #include <cstdlib>  // abort
-#define SYCLACADEMY_ASSERT(cond)                                    \
-  if (!(cond)) {                                                    \
-    std::fprintf(stderr, "%s failed in %s:%d:%s\nExiting\n", #cond, \
-                 __BASE_FILE__, __LINE__, __FUNCTION__);            \
-    std::abort();                                                   \
+#define SYCLACADEMY_ASSERT(cond)                                              \
+  if (!(cond)) {                                                              \
+    std::fprintf(stderr, "[FAILURE] %s failed in %s:%d:%s\nExiting\n", #cond, \
+                 __BASE_FILE__, __LINE__, __FUNCTION__);                      \
+    std::abort();                                                             \
+  } else {                                                                    \
+    std::printf("[SUCCESS] Test passed\n");                                   \
   }
```

Comment:
These are long enough now that we should probably wrap them in `do ... while(0)` to ensure that they don't end up doing the wrong thing if used in certain one-line constructions or similar.

Reply:
It's still a single statement so should be fine, but doesn't hurt to be doubly sure so I added a commit with this change.